### PR TITLE
Update SyncPackages.ps1

### PR DESCRIPTION
**Pull request by donnybell**
Added a try/catch for "Add-Type -AssemblyName System.Net.Http" to prevent the following error:
`New-Object : Cannot find type [System.Net.Http.HttpClientHandler]: verify that the assembly containing this type is loaded.`

### CODE REVIEW

**Comment by donnybell** [comment]
https://octopus.com/docs/octopus-rest-api/examples/feeds/synchronize-packages#script Also needs to be synced with this script once the PR is merged

**Comment by donnybell** [comment]
For lines 22 and 31, should these values be pre-filled or mandatory with no pre-filled value?

**Comment by harrisonmeister** [comment]
> For lines 22 and 31, should these values be pre-filled or mandatory with no pre-filled value?

@donnybell - Unknown what the intentional behavior was supposed to be. It was written by Engineering, so someone who worked on Bento would need to comment. I think the idea is that they were prefilled for convenience since the `Default` space is probably the only known value (since Octopus creates a space called `Default`). That said, it could make users fall into the pit of failure. e.g., If you don't specify a value for destination space, do you _really_ want to copy from/to the same space?

I'm happy if you want to change the behavior you described. You could also add a check to make sure the source and destination space aren't the same values?

File: REST/PowerShell/Feeds/SyncPackages.ps1 [hunk 1]
```diff
@@ -137,6 +137,16 @@ $destinationSpaceId = ((Invoke-RestMethod -Method Get -Uri "$destinationOctopusU
...
 $httpClientTimeoutInMinutes = 60
+Try { 
+    Add-Type -AssemblyName System.Net.Http
+    $client = [System.Net.Http.HttpClient]::new()
+}
+Catch {
+    Write-Warning "Something went wrong when trying the command:"
+    Write-Warning "Add-Type -AssemblyName System.Net.Http"
+    Write-Warning "`$client = [System.Net.Http.HttpClient]::new()"
+    Write-Host "Continuing..."
```
Comment (harrisonmeister) [inline REST/PowerShell/Feeds/SyncPackages.ps1:148]: Should this just continue? What are the impacts of not stopping at this point?

Comment (donnybell) [reply REST/PowerShell/Feeds/SyncPackages.ps1:148]: The next line after Continuing will error with `New-Object : Cannot find type [System.Net.Http.HttpClientHandler]: verify that the assembly containing this type is loaded.`

Maybe this is unfounded, but my thinking was that the only reason why this would fail would be permissions related. You can run the two lines ad nauseam as they generate no output and no issue running the command repeatedly. If `Add-Type -AssemblyName System.Net.Http` were to fail, it is also possible that the `System.Net.Http` may already exist as required to run the rest of the script without issue. I'm not an expert, but my understanding is that this is native in PowerShell Core 6+. I ran into the error in PS5.1 on a Server2022 box.

tl;dr - I'm happy to remove the try/catch if you think it is unnecessary

File: REST/PowerShell/Feeds/SyncPackages.ps1 [hunk 2]
```diff
@@ -137,6 +137,16 @@ $destinationSpaceId = ((Invoke-RestMethod -Method Get -Uri "$destinationOctopusU
...
+}
```
Comment (harrisonmeister) [inline REST/PowerShell/Feeds/SyncPackages.ps1:149]:
```suggestion
if (-not('System.Net.Http.HttpClient' -as [type])) {
    try {
        Write-Warning "System.Net.Http.HttpClient type not found. Trying to load System.Net.Http assembly"
        Add-Type -AssemblyName System.Net.Http
    }
    catch {
        Write-Error "Can't load required System.Net.Http Assembly!"
       exit 1
    }
}
```

Comment (donnybell) [reply REST/PowerShell/Feeds/SyncPackages.ps1:149]: LGTM!

**Review by harrisonmeister — APPROVED** [review]
